src/cmd/browse.rs

```diff
@@ -33,18 +35,34 @@ fn open_url(url: &url::Url) -> Result<()> {
 fn open_url(url: &url::Url) -> Result<()> {
     std::process::Command::new("open")
         .arg(url.to_string())
-        .spawn()?;
+        .spawn()?
+        .wait()?;
 
     Ok(())
 }
 
 #[cfg(all(not(windows), not(target_os = "macos")))]
 fn open_url(url: &url::Url) -> Result<()> {
-    std::process::Command::new("xdg-open")
-        .arg(url.to_string())
-        .spawn()?;
+    // c.f. https://github.com/cli/browser/blob/main/browser_linux.go
+    let commands = ["xdg-open", "x-www-browser", "www-browser", "wslview"];
 
-    Ok(())
+    for command in commands {
+        match std::process::Command::new(command)
+            .arg(url.to_string())
+            .spawn()
+        {
+            Ok(mut child) => {
+                child.wait()?;
+                return Ok(());
+            }
+            Err(e) => match e.kind() {
+                std::io::ErrorKind::NotFound => continue,
+                _ => return Err(e.into()),
+            },
+        }
+    }
+
+    Err(anyhow!("no commands were found to open the url"))
```

Comment:
Let the user know what commands are required to do this in the error mesaage: xdg-open, x-www-browser, www-browser, or wslview.

Reply:
fixed!